rust/connlib/tunnel/src/tests/reference.rs

```diff
@@ -132,9 +136,22 @@ impl ReferenceStateMachine for ReferenceState {
             )
             .with(
                 1,
-                (cidr_resource(8), sample::select(state.all_gateways())).prop_map(
-                    |(resource, gateway)| Transition::AddCidrResource { resource, gateway },
-                ),
+                (cidr_resource(8), sample::select(state.all_gateways()))
+                    .prop_filter(
+                        "tests doesn't support yet CIDR resources overlapping DNS resources",
+                        |(r, _)| {
+                            IpNetwork::from_str(IPV4_RESOURCES)
```

Comment:
Didn't we at some point upstream a const-ctor?

Reply:
I'm waiting on [this](https://github.com/JakubOnderka/ip_network/issues/13), we could patch with `main`(`ip_network_table` uses crates.io version)